expr/interval_compound_literal_test.go

```diff
@@ -0,0 +1,298 @@
+package expr
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/stretchr/testify/assert"
+	"github.com/substrait-io/substrait-go/proto"
+	"github.com/substrait-io/substrait-go/types"
+	"google.golang.org/protobuf/testing/protocmp"
+)
+
+const (
+	yearVal          int32 = 1
+	monthVal         int32 = 10
+	dayVal           int32 = 100
+	secondsVal       int32 = 1000
+	subSecondsVal    int64 = 10000
+	negativeInt32Val int32 = -5
+	negativeInt64Val int64 = -100000
+)
+
+func TestDatePartBuilder(t *testing.T) {
+
+	var origYearVal, origMonthVal, origDayVal, origSecondsVal int32
+	var origSubSecondsVal int64
+	origYearVal = 1
+	origMonthVal = -2
+	origDayVal = 3
+	origSecondsVal = -4
+	origNullabilityVal := types.NullabilityRequired
+	origPrecisionVal := types.PrecisionEMinus7Seconds
+	origSubSecondsVal = 5
+	origLiteral := IntervalCompoundLiteral{Years: origYearVal, Months: origMonthVal, Days: origDayVal, Seconds: origSecondsVal, SubSeconds: origSubSecondsVal, SubSecondPrecision: origPrecisionVal, Nullability: origNullabilityVal}
```

Comment:
expand this into a multi-line literal to make it easier to read please?

Reply:
since remove With* Methods so this test case isn't needed. Removed this test case